hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -378,19 +383,45 @@ protected Option<Pair<String, CompactionOperation>> getPendingLogCompactionOpera
 
   @Override
   protected boolean isPartitionAvailableInStore(String partitionPath) {
-    return partitionToFileGroupsMap.containsKey(partitionPath);
+    fsLock.readLock();
+    try {
+      return partitionToFileGroupsMap.containsKey(partitionPath);
+    } finally {
+      fsLock.readUnlock();
+    }
   }
 
   @Override
   protected void storePartitionView(String partitionPath, List<HoodieFileGroup> fileGroups) {
-    LOG.debug("Adding file-groups for partition :" + partitionPath + ", #FileGroups=" + fileGroups.size());
-    List<HoodieFileGroup> newList = new ArrayList<>(fileGroups);
-    partitionToFileGroupsMap.put(partitionPath, newList);
+    fsLock.writeLock();
```

Comment:
Isn't this already being called under write lock?

Reply:
Adjust to - > assert hasWriteLock